src/test/java/sk/freemap/gpxAnimator/ui/DurationFormatterTest.java

```diff
@@ -0,0 +1,50 @@
+package sk.freemap.gpxAnimator.ui;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class DurationFormatterTest {
+
+    private static DurationFormatter df;
+    private static Map<Long,String> edgeCases = new HashMap<>();
+
+    @BeforeAll
+    static void Setup(){
+
+        df = new DurationFormatter();
+        edgeCases.put(Long.MIN_VALUE,"-106751991167d -7h -12m -55s -808ms");
+        edgeCases.put(Long.MAX_VALUE,"106751991167d 7h 12m 55s 807ms");
+        edgeCases.put(-1L,"0d 0h 0m 0s -1ms");
+        edgeCases.put(0L,"0d 0h 0m 0s 0ms");
+        edgeCases.put(100L,"0d 0h 0m 0s 100ms");
+        edgeCases.put(-100L,"0d 0h 0m 0s -100ms");
+        edgeCases.put(null,"");
+
+    }
```

Comment:
Try do use `@ParameterizedTest`, that is exactly for this use case. If you have not used it before, take a look at the `DurationSpinnerModelTest` which I refactored yesterday. In case of questions, please feel free to ask me.